Tasks/SonarQubePostTest/SummaryReport/QualityGateSection.ps1

```diff
@@ -0,0 +1,222 @@
+
+function GetQualityGateSectionContent
+{
+    Write-Verbose "Formatting the quality gate report section"
+    
+    WaitForAnalysisToFinish
+    
+    $qualityGateResponse = FetchQualityGateDetails
+    $qualityGateStatus = $qualityGateResponse.projectStatus.status
+    $section1 = FormatQualityGateStatusSection $qualityGateStatus
+    
+    $warningsAndErrors = GetQualityGateWarningsAndErrors $qualityGateResponse    
+    $section2 = FormatQualityGateMessagesSection $warningsAndErrors
+    
+    $content = FormatQualityGateSection $section1 $section2
+    return $content
+}
+
+function FormatQualityGateSection
+{
+    param ($section1, $section2)
+    
+    return "$section1 $section2"
+}
+
+function FormatQualityGateMessagesSection
+{
+    param ($messages)
+    
+    if (HasElements $messages)
+    {
+        Write-Host "here"
+        $sb = New-Object "System.Text.StringBuilder"
+        [void]$sb.AppendLine('<table border="0" style="border-top: 1px solid #eee;border-collapse: separate;border-spacing: 0 2px;">')
+        
+        foreach ($message in $messages)
+        {
+            Write-Verbose "Value $($message.actualValue)"
+            
+            [void]$sb.AppendLine("<tr>")
+            [void]$sb.AppendLine("<td><span style=""padding-right:4px;"">$($message.metric_name)</span></td>")
+            [void]$sb.AppendLine("<td style=""text-align: center; background-color:$($message.color); color:#fff;""><span style=""padding:0px 2px"">$($message.actualValue)</span></td>")
+            [void]$sb.AppendLine('<td>&nbsp;' + $message.comparator + ' ' + $message.threshold + '</td>')
+            [void]$sb.AppendLine("</tr>")
+        }
+        
+        [void]$sb.AppendLine("</table>")
+    } 
+    
+    return $sb.ToString()
+}
+
+function FormatQualityGateStatusSection  
+{
+    param ($qualityGateStatus)   
+    
+    $qualityGateColor = "" 
+    $qualityGateLabel = ""
+    GetQualityGateStatusVisualDetails $qualityGateStatus ([ref]$qualityGateColor) ([ref]$qualityGateLabel)
+    
+    $template  = '<div style="padding:5px 0px">
+<span>Quality Gate</span>
+<span style="padding:4px 10px; margin-left: 5px; background-color:{0}; color:#fff; display:inline-block">{1}</span>
+</div>'
+    
+    $reportContents = [String]::Format($template, $qualityGateColor, $qualityGateLabel)
+    return $reportContents
+}
+
+#
+# Fetches the quality gate status and the errors and warnings and also the metrics in order to provide a user friendly report.
+# Note: this is method is unit tested
+#
+function GetQualityGateWarningsAndErrors
```

Comment:
This is the where the "view model" gets constructed based on data from the SQ server